node/pkg/celo/celoimpl.go

```diff
@@ -0,0 +1,263 @@
+// This implements the interface to the "go-ethereum-ish" library used by Celo.
+
+package celo
+
+import (
+	"context"
+
+	celoBind "github.com/celo-org/celo-blockchain/accounts/abi/bind"
+	celoCommon "github.com/celo-org/celo-blockchain/common"
+	celoTypes "github.com/celo-org/celo-blockchain/core/types"
+	celoClient "github.com/celo-org/celo-blockchain/ethclient"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	ethCommon "github.com/ethereum/go-ethereum/common"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+	ethEvent "github.com/ethereum/go-ethereum/event"
+
+	celoAbi "github.com/certusone/wormhole/node/pkg/celo/abi"
+	ethAbi "github.com/certusone/wormhole/node/pkg/ethereum/abi"
+
+	"go.uber.org/zap"
+)
+
+type CeloImpl struct {
+	NetworkName string
+	logger      *zap.Logger
+	client      *celoClient.Client
+	filterer    *celoAbi.AbiFilterer
+	caller      *celoAbi.AbiCaller
+}
+
+func (e *CeloImpl) SetLogger(l *zap.Logger) {
+	e.logger = l
+}
+
+func (e *CeloImpl) DialContext(ctx context.Context, rawurl string) error {
+	var err error
+	e.client, err = celoClient.DialContext(ctx, rawurl)
+	return err
+}
+
+func (e *CeloImpl) NewAbiFilterer(address ethCommon.Address) error {
+	var err error
+	e.filterer, err = celoAbi.NewAbiFilterer(celoCommon.BytesToAddress(address.Bytes()), e.client)
+	return err
+}
+
+func (e *CeloImpl) NewAbiCaller(address ethCommon.Address) error {
+	var err error
+	e.caller, err = celoAbi.NewAbiCaller(celoCommon.BytesToAddress(address.Bytes()), e.client)
+	return err
+}
+
+func (e *CeloImpl) GetCurrentGuardianSetIndex(ctx context.Context) (uint32, error) {
+	if e.caller == nil {
+		panic("caller is not initialized!")
+	}
+
+	opts := &celoBind.CallOpts{Context: ctx}
+	return e.caller.GetCurrentGuardianSetIndex(opts)
+}
+
+func (e *CeloImpl) GetGuardianSet(ctx context.Context, index uint32) (ethAbi.StructsGuardianSet, error) {
+	if e.caller == nil {
+		panic("caller is not initialized!")
+	}
+
+	opts := &celoBind.CallOpts{Context: ctx}
+	celoGs, err := e.caller.GetGuardianSet(opts, index)
```

Comment:
You need to check this err btw. The [errcheck](https://github.com/kisielk/errcheck) linter will find these sorts of things for you automatically.

Reply:
Sticking with the panics, since these errors would be programming bugs, rather than legitimate runtime errors.

Added that error check.